test/functional/ex_cmds/excmd_spec.lua

```diff
@@ -50,4 +52,22 @@ describe('Ex cmds', function()
     eq(2, funcs.exists(':defer'))
     eq('defer', funcs.fullcommand('defer'))
   end)
+
+  it(':.! should pipe line buffer to &shell', function()
```

Comment:
these tests would fit best in `test/functional/vimscript/system_spec.lua` , here: https://github.com/neovim/neovim/blob/28f1f65bb30768db3423dc6eb39b2d75dfaba326/test/functional/vimscript/system_spec.lua#L632

Reply:
Right, I will move them here ! Thanks for pointing me to this file.